### PR TITLE
update vm with chef extension template to api 05-01-2015

### PR DESCRIPTION
**Pull request by kmouss**
Update Windows vm with chef extension template to api 05-01-2015

### CODE REVIEW

File: chef-extension-windows-vm/azuredeploy-parameters.json [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    "vmExtensionName": {
+      "value": "ChefClient"
+    },
+    "client_rb": {
+	"value": ""
+    },
+    "runlist": {
+	"value": ""
+    },
+    "validation_key: {
```
Comment (singhkay) [inline chef-extension-windows-vm/azuredeploy-parameters.json]: missing closing quote

**Comment by singhkay** [comment]
lgtm @johngossman @kenazk

File: chef-extension-windows-vm/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+                        {
+                            "id": "[resourceId('Microsoft.Network/networkInterfaces',variables('nicName'))]"
+                        }
+                    ]
+                }
+            }
+        },
+  {
+    "type": "Microsoft.Compute/virtualMachines/extensions",
+    "name": "[concat(parameters('vmName'),'/', parameters('vmExtensionName'))]",
```
Comment (johngossman) [inline chef-extension-windows-vm/azuredeploy.json]: There is no need to parameterize ExtensionName. Just hardcode it to "chefExtension"

File: chef-extension-windows-vm/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+                "2012-Datacenter",
+                "2012-R2-Datacenter",
+                "Windows-Server-Technical-Preview"
+            ],
+            "metadata": {
+                "description": "The Windows version for the VM. This will pick a fully patched image of this given Windows version. Allowed values: 2008-R2-SP1, 2012-Datacenter, 2012-R2-Datacenter, Windows-Server-Technical-Preview."
+            }
+        },
+
+    	"vmExtensionName": {
```
Comment (johngossman) [inline chef-extension-windows-vm/azuredeploy.json]: Doesn't need to be a parameter

File: chef-extension-windows-vm/azuredeploy.json [hunk 4]
```diff
@@ -0,0 +1,242 @@
...
+            }
+        },
+        "location": {
+            "type": "String",
+            "defaultValue" : "West US",
+            "metadata": {
+                 "Description": "Azure region where the VM will be deployed"
+            }
+        },
+        "vmSize": {
```
Comment (johngossman) [inline chef-extension-windows-vm/azuredeploy.json]: Added allowedValues as we do in the 101 examples.

File: chef-extension-windows-vm/azuredeploy.json [hunk 5]
```diff
@@ -0,0 +1,242 @@
...
+        },
+        "vmName": {
+            "type": "string",
+	    "defaultValue" : "myVM",
+            "metadata": {
+                 "Description": "Vitual machine's unique name; which will be used in the DSN address"
+            }
+        },
+        "location": {
+            "type": "String",
```
Comment (johngossman) [inline chef-extension-windows-vm/azuredeploy.json]: "string" not "String"

File: chef-extension-windows-vm/azuredeploy.json [hunk 6]
```diff
@@ -0,0 +1,242 @@
...
+            "defaultValue" : "West US",
```
Comment (johngossman) [inline chef-extension-windows-vm/azuredeploy.json]: Added allowedValues as we do in the 101 examples

File: chef-extension-windows-vm/azuredeploy.json [hunk 7]
```diff
@@ -0,0 +1,242 @@
...
+            "type": "string",
+            "metadata": {
+                  "description": "Unique DNS Name for the Public IP used to access the Virtual Machine."
+            }
+        },
+        "vmName": {
+            "type": "string",
+	    "defaultValue" : "myVM",
+            "metadata": {
+                 "Description": "Vitual machine's unique name; which will be used in the DSN address"
```
Comment (johngossman) [inline chef-extension-windows-vm/azuredeploy.json]: Not used in the DNS name. Note required to be a parameter.

File: chef-extension-windows-vm/azuredeploy.json [hunk 8]
```diff
@@ -0,0 +1,242 @@
...
+                        }
+                    }
+                ]
+            }
+        },
+        {
+            "apiVersion": "2015-05-01-preview",
+            "type": "Microsoft.Network/networkInterfaces",
+            "name": "[variables('nicName')]",
+            "location": "[variables('location')]",
```
Comment (johngossman) [inline chef-extension-windows-vm/azuredeploy.json]: location should use the parameter.

File: chef-extension-windows-vm/azuredeploy.json (hunk 7 above)
Comment (kmouss) [inline chef-extension-windows-vm/azuredeploy.json]: made other update, I see this one used here. Or do you mean to not specify an explicit DNS in this VM template.
        {
            "apiVersion": "2015-05-01-preview",
            "type": "Microsoft.Network/publicIPAddresses",
            "name": "[variables('publicIPAddressName')]",
            "location": "[parameters('location')]",
            "properties": {
                "publicIPAllocationMethod": "[variables('publicIPAddressType')]",
                "dnsSettings": {
                    "domainNameLabel": "[parameters('dnsNameForPublicIP')]"
                }
            }

Comment (johngossman) [inline chef-extension-windows-vm/azuredeploy.json]: Sorry vmName: "Vitual machine's unique name; which will be used in the DSN address"

vmName is not part of the DNS address

Comment (kmouss) [inline chef-extension-windows-vm/azuredeploy.json]: Got it. fixed and also noticed the vmName was both variable and param, so I fix that too - thanks

File: chef-extension-windows-vm/azuredeploy.json [hunk 9]
```diff
@@ -0,0 +1,233 @@
...
+            "type": "string",
+            "metadata": {
+                  "description": "Unique DNS Name for the Public IP used to access the Virtual Machine."
+            }
+        },
+        "vmName": {
+            "type": "string",
+	    "defaultValue" : "myVM",
+            "metadata": {
+                 "Description": "Vitual machine's unique name."
```
Comment (singhkay) [inline chef-extension-windows-vm/azuredeploy.json]: Can you make this and "Description" below lowercase?

**Pull request merged**